### PR TITLE
new(engine): raise warning instead of error on not-unique exceptions names

### PR DESCRIPTION
**Pull request by mrgian**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:
Makes the loader raise a warning instead of an error when a rule is defining exceptions with the same name

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(engine): raise warning instead of error on not-unique exceptions names
```

### CODE REVIEW

File: userspace/engine/rule_loader_reader.cpp [hunk 1]
```diff
@@ -319,8 +319,10 @@ static void read_rule_exceptions(
 		// Check if an exception with the same name has already been defined
 		for (auto &exception : exceptions)
 		{
-			THROW((v_ex.name == exception.name),
-			       "Exceptions names in the same object must be unique", ex_ctx);
+			if(v_ex.name == exception.name)
+			{
+				cfg.res->add_warning(falco::load_result::LOAD_EXCEPTION_NAME_NOT_UNIQUE, "Multiple exceptions defined with the same name", ex_ctx);
```
Comment (jasondellaluce) [inline userspace/engine/rule_loader_reader.cpp:324]:
```suggestion
				cfg.res->add_warning(falco::load_result::LOAD_EXCEPTION_NAME_NOT_UNIQUE, "Multiple definitions of exception '" + v_ex.name + "' in the same rule", ex_ctx);
```
Suggesting a more semantic error.

File: userspace/engine/falco_engine_version.h [hunk 2]
```diff
@@ -20,7 +20,7 @@ limitations under the License.
 
 // The version of this Falco engine
 #define FALCO_ENGINE_VERSION_MAJOR 0
-#define FALCO_ENGINE_VERSION_MINOR 39
+#define FALCO_ENGINE_VERSION_MINOR 40
```
Comment (jasondellaluce) [inline userspace/engine/falco_engine_version.h:23]: I think we can avoid bumping the engine in this case. This is effectively a follow-up to the same change we made few days ago.

**Review by jasondellaluce — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**